lib/services/service_brokers/v2/client.rb

```diff
@@ -94,21 +94,22 @@ def create_service_key(key, arbitrary_parameters: {})
     end
 
     def bind(binding, arbitrary_parameters: {}, accepts_incomplete: false)
-      path              = service_binding_resource_path(binding.guid, binding.service_instance.guid, accepts_incomplete: accepts_incomplete)
-      body              = {
+      path = service_binding_resource_path(binding.guid, binding.service_instance.guid, accepts_incomplete: accepts_incomplete)
+      key_required_parameters = { credential_client_id: @config.get(:cc_service_key_client_name) } if binding.is_a?(VCAP::CloudController::ServiceKey)
```

Comment:
This line made me realise that the abstraction of the `required_parameters` as a method in the model might not quite right if we cant use it for all three cases. Do you see any issues with accessing the config in the ServiceKey model?

```
def required_parameters
      { credential_client_id: VCAP::CloudController::Config.config.get(:cc_service_key_client_name) }
end
```

That would remove the need for the client to know the specific of the Service Key

Reply:
I don't see any issues, it's just a pattern that we don't seem to be following elsewhere. It seems to me weird that a model class can be configured in any way. I also didn't like the required_parameters in the model, as that means the model does know something about OSBAPI requests, which breaks encapsulation. It is cleaner for me to leave those things to the client. Our problem is that our client is a big ball of mud.